src/HasStatuses.php

```diff
@@ -45,17 +45,33 @@ public function isValidStatus(string $name, ?string $reason = null): bool
      */
     public function latestStatus(...$names): ?Status
     {
-        $names = is_array($names) ? Arr::flatten($names) : func_get_args();
-
         $statuses = $this->relationLoaded('statuses') ? $this->statuses : $this->statuses();
 
+        $names = is_array($names) ? Arr::flatten($names) : func_get_args();
         if (count($names) < 1) {
             return $statuses->first();
         }
 
         return $statuses->whereIn('name', $names)->first();
     }
 
+    public function hasStatus($name): bool
```

Comment:
Let's make this method name more explicted. `hasStatus` sound as if it only check the current status. Rename this to `hasEverHadStatus` (or something better)